### PR TITLE
updated tests/implementations to work with hapi 17

### PR DESCRIPTION
**Pull request by AVVS**
Did some refactoring, all tests pass, though I'd clean up code a bit

### CODE REVIEW

**Comment by metoikos** [comment]
I've tested with Twitter in a live app and worked like a charm.

File: examples/arcgisonline.js [hunk 1]
```diff
@@ -1,46 +1,39 @@
...
-    path: '/bell/door',
-    config: {
-        auth: 'arcgisonline',
-        handler: function (request, h) {
-
-            reply('<pre>' + JSON.stringify(request.auth.credentials, null, 4) + '</pre>');
+    server.auth.strategy('arcgisonline', 'bell', {
+        provider: 'arcgisonline',
+        password: 'cookie_encryption_password_secure',
+        isSecure: false,
```
Comment (benatkin) [inline examples/arcgisonline.js]: This should be true for the purposes of an example IMHO, with a comment that it should only be set to false for development.

**Comment by benatkin** [comment]
I got GitHub OAuth working with this. 👍🏻

File: examples/arcgisonline.js (hunk 1 above)
Comment (AVVS) [reply examples/arcgisonline.js]: Well, I merely updated to async/await, and did some linting (ie top-level await is not yet possible, hence the wrap)

**Comment by wraithgar** [comment]
I've got this working happy-path with quite a few providers, including a custom one.  However there is a change between this branch and the current bell behavior that I think is worth bringing up.

If the request that bell makes to the `temporary` url (and probably others, haven't tested those) has an error, the response code from that request will bubble up to the client making the request.

The behavior used to be (correctly I would think) that a 500 error would be sent back to the client.

**Comment by marpstar** [comment]
FWIW, I tested this with LinkedIn tonight and it worked great.

**Comment by qraynaud** [comment]
Working great with Google too. Thanks for the hard work!

**Comment by flipxfx** [comment]
Any timeline on when this will be merged? Can't use hapi 17 without bell.

**Comment by qraynaud** [comment]
I think that @wraithgar is correct about the fact that an error generated by a provider should result on an error 500 and not bubble up to the client.

What are your thoughts on this @AVVS ? And are you willing to fix this if you are in agreement over this ?

**Pull request closed without merging**